File: Java/Logic-1/shareDigit.java

public class logic1 {

	// given two ints, each in the range 10..99
	// return true if there is a digit that appears in both numbers, such as the 2 in 12 and 23
	public boolean shareDigit(int a, int b) {
		
		boolean bool = false;
		
		if ((a >= 10 && a <= 99) && (b >= 10 && b <= 99)) {
			if ((a / 10 == b / 10 || a / 10 == b % 10) || (a % 10 == b / 10 || a % 10 == b % 10 ))
				bool = true;
		}
			
		return bool;
		
	}
	
	public static void main(String[] args) {
		
		logic1 o = new logic1();
		
		System.out.println(o.shareDigit(12, 23));
		System.out.println(o.shareDigit(12, 43));
		System.out.println(o.shareDigit(12, 44));

	}

}

File: Python/Logic-1/share_digit.py

# given two ints, each in the range 10..99
# return true if there is a digit that appears in both numbers, such as the 2 in 12 and 23
def share_digit(a, b):
    bool = False

    if (a >= 10 and a <= 99) and (b >= 10 and b <= 99):
        if (int(a/10) == int(b/10) or int(a/10) == b%10) or (a%10 == int(b/10) or a % 10 == b % 10):
            bool = True

    return bool

print(share_digit(12, 23))
print(share_digit(12, 43))
print(share_digit(12, 44))
